### PR TITLE
Add three new uniques

### PR DESCRIPTION
**Pull request by PLynx01**
I would like to add three new conditionals. These are

<is controlled by Human>
<is controlled by AI>
Sell [buildingFilter] [cityFilter]

The first two are very simple. They will be used by modders to make features restricted to Human players or AI.
The last is the variation of removeBuilding unique was added by me. The difference is that SellBuilding gives money for removed buildings. 

I hope the implementation would be quicker than in case of my before PR.

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt [hunk 1]
```diff
@@ -598,6 +599,9 @@ enum class UniqueType(val text: String, vararg targets: UniqueTarget, val flags:
 
 
     /////// civ conditionals
+    ConditionalIsHuman("is controlled by human", UniqueTarget.Conditional),
+    ConditionalIsAI("is controlled by AI", UniqueTarget.Conditional),
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/unique/UniqueType.kt:603]: This is not how conditionals are phrased - they need to start with "when ..."

File: core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt [hunk 2]
```diff
@@ -664,6 +664,29 @@ object UniqueTriggerActivation {
...
 
+            UniqueType.SellBuilding -> {
+
+                val applicableCities = civInfo.cities.asSequence().filter {
+                    it.matchesFilter(unique.params[1])
+                }
+
+                for (applicableCity in applicableCities) {
+                    var buildingsToSell = applicableCity.cityConstructions.getBuiltBuildings().filter {
+                        it.matchesFilter(unique.params[0])
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt:675]: Why not apply both filters here? Filter for isSellable as well
And you don't need to .toSet() this at all

**Comment by PLynx01** [comment]
I've applied your recommendations. Can you merge now?

File: core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt (hunk 2 above)
Comment (yairm210) [reply core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt:675]: I think you misunderstood me... .filter { A && B }

Comment (PLynx01) [reply core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt]: I've updated this function according to your remark. Is this PR ready to be merged?

**Comment by SeventhM** [comment]
Question: why is human/AI checks conditionals and not nationFilters?

**Comment by yairm210** [comment]
@SeventhM GOOD POINT!

**Comment by yairm210** [comment]
Nation filters should be "AI" and "Human"

**Comment by PLynx01** [comment]
> Nation filters should be "AI" and "Human"

So do I need to modify the nationFilter? I mean, add human and AI?

**Comment by PLynx01** [comment]
> Nation filters should be "AI" and "Human"

So, how the mentioned uniques should look like?

Option A
(refers to current nation)
<when controlled by human>
<when controlled by AI>

Option B
<when [nationFilter] controlled by human>
<when [nationFilter] controlled by AI>

Option A, B, both or maybe something else?
Give me the text of the proposed uniques if you can.

**Comment by yairm210** [comment]
Not as a unique, but as an option for nationFilter
So nationFilter should accept "AI" and "Human" as acceptable filters and filter them

**Comment by PLynx01** [comment]
> Not as a unique, but as an option for nationFilter So nationFilter should accept "AI" and "Human" as acceptable filters and filter them

In that case, we don't need new unique. We need to add AI and Human options to this:
<for [nationFilter]>

**Comment by PLynx01** [comment]
@yairm210 
Is this what you asked for?

13e1871

**Comment by PLynx01** [comment]
@yairm210 

Is my PR ready to be merged? Or do you have some remarks about it.

**Comment by yairm210** [comment]
This raises a good point - AI vs Human is not actually *nation-dependant*, it's actually *civ dependant*. We currently don't have a CivFilter to contain both Nation filters and other filters, makes me wonder what other "civ but not nation" filters we're missing out on...
We'll need to meditate on this

**Comment by PLynx01** [comment]
> This raises a good point - AI vs Human is not actually *nation-dependant*, it's actually *civ dependant*. We currently don't have a CivFilter to contain both Nation filters and other filters, makes me wonder what other "civ but not nation" filters we're missing out on...
> We'll need to meditate on this 

Can't we just make two simple Human/AI uniques as I stated in beginning of the thread?

What is the reason to integrate them into <for [nationFilter]> ?

Please explain me

**Comment by yairm210** [comment]
So they can be used anywhere that we have unit filters or nation filters

**Comment by PLynx01** [comment]
One more question:
What is the difference between Nation and Civilization in game?

Can these two be used interchangeably?

**Comment by SeventhM** [comment]
> What is the difference between Nation and Civilization in game

In the code, a "nation" is the base civ and its uniques, while a "civilization" is more or less a collection that includes the player and all of its mechanics. In game... I think so far it's more or less interchangeable. I don't think we actually have anything where information on the nation itself matters besides I guess the civilopedia

**Comment by SeventhM** [comment]
> only one use

mapUnitFilter and tileFilter also uses nationFilter

**Comment by PLynx01** [comment]
So, what's our decision?

1. Altering the nationFilter to accept Human/AI 

2. Separate two uniques for Human/AI players

3. Other solution (please reply)

**Comment by yairm210** [comment]
I Implemented solution 1

**Comment by PLynx01** [comment]
> I Implemented solution 1 

If the Human/AI filter is implemented, should we add the Sell Building(s) unique mentioned earlier?

**Comment by yairm210** [comment]
Sell building can be added as a unique

**Comment by PLynx01** [comment]
> Sell building can be added as a unique

So what should I do to implement this?

**Comment by PLynx01** [comment]
The discussion about SellBuildings has been moved to #10514

**Pull request closed without merging**